### ISSUE TITLE
computational efficiency 

### ISSUE THREAD

**Issue by dbuscombe-usgs** [dbuscombe-usgs/segmentation_zoo#13]
a memory bottleneck in multiclass problems is caused by the use of one-hot encoded label stacks

binary problems can use larger TARGET_SIZE and / or BATCH_SIZE than multiclass problems, for this reason

One possible workaround is to train separate binary models for each class in a multiclass problem

**Comment by ebgoldstein** [comment]
another idea is to use label encoding and: don't use softmax on the last layer, use `SparseCategoricalCrossentropy` as loss with `from_logits=True`

**Comment by dbuscombe-usgs** [comment]
- [ ] merge branch containing distributed training 
- [ ] implement sparse loss (requires flattened 2d labels, not one-hot encoded)
